lib/model/model.go

```diff
@@ -1214,9 +1192,9 @@ func (m *model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 
 	// Assemble the device information from the connected device about
 	// themselves and us for all folders.
-	ccDeviceInfos := make(map[string]*indexSenderStartInfo, len(cm.Folders))
+	ccDeviceInfos := make(map[string]*indexHandlerStartInfo, len(cm.Folders))
```

Comment:
Similarly here I wonder if we could avoid building and carrying around a bunch of `indexHandlerStartInfo`s and instead just tell the registry about them at the point were we would now build the startinfo?

Reply:
That type is badly named, I renamed it to `clusterConfigDeviceInfo`: It isn't exclusively used for index handling, it also contains the encryption info, is used for auto-accepting folders, ...  
We can't push the info immediately to the index-handler and delegate hashing out what to do with it directly, as it would e.g. create a brief race condition between sending indexes and checking encryption later in `ClusterConfig` (which might need to prevent the connection). Generally the index-handler would have to check for the same conditions that are already checked in `ccHandleFolders`, so I would expect more duplication than simplification there.  
Or maybe I am just too deep into all of this and need to get some distance to grasp the bigger picture.